src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -425,25 +423,22 @@ public override void SetBufferSize(int bufferSize)
         /// <returns>SNI error code</returns>
         public override uint Send(SNIPacket packet)
         {
-            lock (this)
```

Comment:
Why is it ok to remove these locks?

Reply:
I see you already answered this at https://github.com/dotnet/corefx/pull/27187#issuecomment-366135297. Thanks